include/cuda/std/detail/libcxx/include/__iterator/reverse_iterator.h

```diff
@@ -0,0 +1,186 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___ITERATOR_REVERSE_ITERATOR_H
+#define _LIBCUDACXX___ITERATOR_REVERSE_ITERATOR_H
+
+#ifndef __cuda_std__
+#include <__config>
+#include <__iterator/iterator.h>
+#include <__iterator/iterator_traits.h>
+#include <__memory/addressof.h>
+#include <__type_traits/void_t.h>
+#else
+#include "../__iterator/iterator.h"
+#include "../__iterator/iterator_traits.h"
+#include "../__memory/addressof.h"
+#include "../__type_traits/void_t.h"
+#endif // __cuda_std__
+
+#if defined(_LIBCUDACXX_USE_PRAGMA_GCC_SYSTEM_HEADER)
+#pragma GCC system_header
+#endif
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+template <class _Tp, class = void>
+struct __is_cpp17_stashing_iterator : false_type {};
+
+template <class _Tp>
+struct __is_cpp17_stashing_iterator<_Tp, __void_t<typename _Tp::__stashing_iterator_tag>>
+  : true_type {};
+
+template <class _Iter>
+class _LIBCUDACXX_TEMPLATE_VIS reverse_iterator
+    : public iterator<typename iterator_traits<_Iter>::iterator_category,
+                      typename iterator_traits<_Iter>::value_type,
+                      typename iterator_traits<_Iter>::difference_type,
+                      typename iterator_traits<_Iter>::pointer,
+                      typename iterator_traits<_Iter>::reference>
+{
+private:
+    /*mutable*/ _Iter __t;  // no longer used as of LWG #2360, not removed due to ABI break
```

Comment:
Not really all that relevant to this PR, but since you've made me look through all this code: I wonder why this isn't changed with unstable ABI and if we should bite the bullet and remove it under unstable ABI in our copy.

Reply:
We should consider that in the future, but not in this PR